server/src/api/modules/auth.js

```diff
@@ -47,7 +47,7 @@ export const login = (req, res) => {
       if (!oldUser) return res.status(404).send("User does not exist")
       oldUser.comparePassword(password, (err, isMatch) => {
         if (err) {
-          return res.status(401).send("Unauthorized")
+          return res.status(500).send("Error checking use password")
```

Comment:
Let's leave the status code as `401`. `Unauthorized` is a more appropriate error for a wrong password than `Internal Server Error`.

But I agree let's send a more detailed message back. Perhaps something more to point? `Wrong password` or `Password does not match`

Reply:
This error comes back from `comparePassword` does not indicate password match or unmatch. It is the error of `comparePassword` function itself.

According to your code, the case `unmatched password` is already handled below.
```
if (err) {
          return res.status(500).send("Error checking use password")
        }
        if (isMatch) {
          // let token = generateToken(user)
          const token = jwt.sign({ id: oldUser._id }, JWT_SECRET, {
            expiresIn: 86400 // 24 hours
          })

          const payload = { user: oldUser, token }
          res.status(200).json(payload)
        } else {
          return res.status(401).send("Invalid password") <------ HERE
        }
```

Reply:
@alexbotello